build_and_deploy/utils/arm_template_utils.ts

```diff
@@ -150,16 +168,23 @@ function replaceVariables(armTemplate: string): string {
     return armTemplate;
 }
 
-function replaceStrByRegex(str: string, regex: any): string {
-    return str.replace(regex, function (matchedStr: string, arg1: string, arg2: string) {
-        if (arg1.endsWith("'")) {
-            arg1 = arg1.substring(1, arg1.length - 1);
-        }
-        if (arg2.startsWith("'")) {
-            arg2 = arg2.substring(1, arg2.length - 1);
-        }
-        return `${arg1}${arg2}`;
+function replaceStrByRegex(str: string): string {
```

Comment:
Can you leave a few comments explaining the functions logic?

Reply:
Sure will add.